src/framework/global/serialization/xmlstreamreader.cpp

```diff
@@ -458,7 +458,11 @@ double XmlStreamReader::readDouble(bool* ok)
 
 int64_t XmlStreamReader::lineNumber() const
 {
-    return m_xml->doc.ErrorLineNum();
+    int64_t lineNum = m_xml->doc.ErrorLineNum();
+    if (lineNum == 0 && m_xml->node) {
+        lineNum = m_xml->node->GetLineNum();
```

Comment:
try to return a more helpful line number based on the most recently processed node